miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BarcodableSearchRestController.java

```diff
@@ -0,0 +1,72 @@
+package uk.ac.bbsrc.tgac.miso.webapp.controller.rest;
+
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isStringEmptyOrNull;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import uk.ac.bbsrc.tgac.miso.dto.BarcodableDto;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.service.BarcodableViewService;
+
+@Controller
+@RequestMapping("/rest/barcodable")
+public class BarcodableSearchRestController extends RestController {
+  private static final String baseUrl = "/miso";
+
+  @Autowired
+  private BarcodableViewService barcodableViewService;
+
+  @RequestMapping(value = "/search", method = RequestMethod.GET)
+  @ResponseBody
+  public List<BarcodableDto> search(@RequestParam(name = "q", required = false) String query) throws IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)